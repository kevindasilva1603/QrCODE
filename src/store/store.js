import { configureStore } from "@reduxjs/toolkit";
import userReducer from "./userSlice"; // Assurez-vous que le chemin est correct

export const store = configureStore({
    reducer: {
        user: userReducer, // Utilisation du userReducer import√©
    },
});
