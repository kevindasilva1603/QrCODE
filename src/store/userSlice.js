import { createSlice } from "@reduxjs/toolkit";

export const userSlice = createSlice({
    name: "user",
    initialState: {
        name: "",
        surname: "",
    },
    reducers: {
        setUser: (state, action) => {
            state.name = action.payload.name;
            state.surname = action.payload.surname;
        },
    },
});

export const { setUser } = userSlice.actions;
export default userSlice.reducer;
