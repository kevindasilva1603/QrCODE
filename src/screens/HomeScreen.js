import React, { useState } from "react";
import { View, TextInput, Button, StyleSheet, Alert } from "react-native";
import { useDispatch } from "react-redux";
import { setUser } from "../store/userSlice";
import { useNavigation } from "@react-navigation/native";

const HomeScreen = () => {
    const [name, setName] = useState("");
    const [surname, setSurname] = useState("");
    const dispatch = useDispatch();
    const navigation = useNavigation();

    const handleGenerateQR = () => {
        if (name && surname) {
            dispatch(setUser({ name, surname }));
            navigation.navigate("QRScreen");
        } else {
            Alert.alert("Erreur", "Veuillez remplir les deux champs.");
        }
    };

    return (
        <View style={styles.container}>
            <TextInput
                style={styles.input}
                placeholder='Nom'
                value={name}
                onChangeText={setName}
            />
            <TextInput
                style={styles.input}
                placeholder='Prénom'
                value={surname}
                onChangeText={setSurname}
            />
            <Button title='Générer le QR Code' onPress={handleGenerateQR} />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
        padding: 16,
        backgroundColor: "#0E0E2C",
    },
    input: {
        width: "80%",
        padding: 12,
        marginVertical: 10,
        borderWidth: 1,
        borderColor: "white",
        borderRadius: 10,
        color: "white",
        backgroundColor: "#1C1C3C",
    },
});

export default HomeScreen;
