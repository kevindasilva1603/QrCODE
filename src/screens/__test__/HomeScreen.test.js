// src/screens/__tests__/HomeScreen.test.js
import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import HomeScreen from "../HomeScreen";
import { Provider } from "react-redux";
import { store } from "../../store/store";
import { NavigationContainer } from "@react-navigation/native";

const mockNavigate = jest.fn();

jest.mock("@react-navigation/native", () => ({
    ...jest.requireActual("@react-navigation/native"),
    useNavigation: () => ({
        navigate: mockNavigate,
    }),
}));

describe("HomeScreen", () => {
    const renderScreen = () =>
        render(
            <Provider store={store}>
                <NavigationContainer>
                    <HomeScreen />
                </NavigationContainer>
            </Provider>
        );

    test("1. Rendu du snapshot", () => {
        const screen = renderScreen();
        expect(screen).toMatchSnapshot();
    });

    test("2. Le composant HomeScreen existe", () => {
        const { getByPlaceholderText, getByText } = renderScreen();
        expect(getByPlaceholderText("Nom")).toBeTruthy();
        expect(getByPlaceholderText("Prénom")).toBeTruthy();
        expect(getByText("Générer le QR Code")).toBeTruthy();
    });

    test("3. Champs de saisie pour nom et prénom existent", () => {
        const { getByPlaceholderText } = renderScreen();
        const inputName = getByPlaceholderText("Nom");
        const inputSurname = getByPlaceholderText("Prénom");
        expect(inputName).toBeTruthy();
        expect(inputSurname).toBeTruthy();
    });

    test("4. Le bouton permettant de générer le QR Code existe", () => {
        const { getByText } = renderScreen();
        const button = getByText("Générer le QR Code");
        expect(button).toBeTruthy();
    });

    test("5. Navigation appelée une fois lors du clic sur le bouton", () => {
        const { getByText, getByPlaceholderText } = renderScreen();

        fireEvent.changeText(getByPlaceholderText("Nom"), "Jean");
        fireEvent.changeText(getByPlaceholderText("Prénom"), "Marc");
        fireEvent.press(getByText("Générer le QR Code"));

        expect(mockNavigate).toHaveBeenCalledTimes(1);
        expect(mockNavigate).toHaveBeenCalledWith("QRScreen");
    });
});
