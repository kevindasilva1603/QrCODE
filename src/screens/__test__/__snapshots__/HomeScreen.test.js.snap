// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen 1. Rendu du snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0E0E2C",
      "flex": 1,
      "justifyContent": "center",
      "padding": 16,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Nom"
    style={
      {
        "backgroundColor": "#1C1C3C",
        "borderColor": "white",
        "borderRadius": 10,
        "borderWidth": 1,
        "color": "white",
        "marginVertical": 10,
        "padding": 12,
        "width": "80%",
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Prénom"
    style={
      {
        "backgroundColor": "#1C1C3C",
        "borderColor": "white",
        "borderRadius": 10,
        "borderWidth": 1,
        "color": "white",
        "marginVertical": 10,
        "padding": 12,
        "width": "80%",
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Générer le QR Code
      </Text>
    </View>
  </View>
</View>
`;
